app/src/main/java/com/ecosystem/kin/app/App.java

```diff
@@ -34,12 +37,30 @@ public void onCreate() {
         }
 
         try {
-            Kin.start(getApplicationContext(), signInData);
+            Kin.start(getApplicationContext(), signInData)
+            .addNativeOfferCallback(getNativeOfferCallback());
         } catch (InitializeException e) {
             e.printStackTrace();
         }
     }
 
+    private Callback<NativeSpendOffer> getNativeOfferCallback() {
+        return new Callback<NativeSpendOffer>() {
+            @Override
+            public void onResponse(NativeSpendOffer response) {
+                // Do checks on the offer
+                Intent intent = new Intent(getApplicationContext(), NativeOfferActivity.class);
```

Comment:
maybe pass response to NativeOfferActivity - to show details of the clicked offer

Reply:
If we want to check adding 2 offers and press on one of them